### ISSUE TITLE
Follow up + checklist

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/textrecipes#12]
Hello @topepo, I'm checking in again if that is okay with you :)

I got a functional (not fully complete, e.i. minimal documentation/checks) version of most of the steps I was thinking about adding first time around (`step_hashfeature` is developed but not quite ready to be pushed.)

Now my next work will be 

- [x] improving documentation
- [x] add vignette
- [x] add print functions
- [x] add tidy functions (tidy.step_*)
- [x] update readme with working example
- [x] check for tokenization https://github.com/tidymodels/recipes/issues/198#issuecomment-422822227
- [x] checks in general

I have a question for you before I start on those. About the `print.*` and `tidy.step_*` do you have a step that I can take inspiration from? (i.e. a print. and tidy. method that you consider done and most up to date with your vision.)

**Comment by topepo** [comment]
Great! Let me know if you have more questions. 

> do you have a step that I can take inspiration from?

There are a ton of examples in `recipes`. Here's a fairly standard one for `step_scale`. 

```r
# widths below are estimates on width of the `cat` text below. 
print.step_scale <-
  function(x, width = max(20, options()$width - 30), ...) {
    # short description of the operation
    cat("Scaling for ", sep = "")
    # This will print the actual terms after `bake` (via the `sds` object in that step)
    # and the raw input beforehand (from `terms`). 
    printer(names(x$sds), x$terms, x$trained, width = width)
    invisible(x)
  }
``` 

For testing, I just wrap a `expect_output` around a call to make sure that they work. 

For `tidy`, you want to return the information that the users might want in a standardized format (tibble). Try to make the column names include `terms` for the selected variables and `values` for anything that they might want. There is some wiggle room here though. Make sure that you export this function. An example:

```r
#' @rdname step_scale
#' @param x A `step_scale` object.
#' @export
tidy.step_scale <- function(x, ...) {
  if (is_trained(x)) {
    res <- tibble(terms = names(x$sds),
                  value = x$sds)
  } else {
    term_names <- sel2char(x$terms)
    res <- tibble(terms = term_names,
                  value = na_dbl) # use rlang type-specific missing values
  }
  res
}
```

I'll try to take a look at `textrecipes` in the next few days.

**Comment by EmilHvitfeldt** [comment]
Okay perfect! I'll try to get a lot done today so you get a more clear view of the package.

I have one more question. All of your steps include the two following

```r
#' @keywords datagen 
#' @concept preprocessing encoding
```

Could you explain the importance and use case of these lines?

**Comment by topepo** [comment]
`keywords` is fairly old-school and almost useless. Its inclusion was probably just muscle-memory. The only exception is using the [`internal`](https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html) keyword. 

*concepts* could be anything really. I put those here as a test bed for some ideas that I have about eventually reorganizing R's documentation. So if you wanted to find _all_ functions related to encodings, that data would be in the Rd files. 

tl;dr

You don't need either of these.

**Comment by EmilHvitfeldt** [comment]
Okay another update! `textrecipes` is at a pretty good spot right now. It has a small collection of steps that that combined should be able to do like 70-80% of what people want. e.i. it is basically ready to use.

Only thing is that `step_hashing`, `step_tf` and `step_tfidf` are a little slow. they are working! but are leaving a little to be desired. I'll see what I can do to speed it up. I'll find some bottlenecks, maybe offload the work to Tensorflow or something. But they are working as intended (just not as fast as I want them to)!

**Comment by topepo** [comment]
The package looks great! I had some issues inside of `recipes` with steps exporting list columns. I'm fixing that now and merge a PR to fix. 

My comments are related to arguments mostly. Take a look at [this page](https://tidymodels.github.io/model-implementation-principles/standardized-argument-names.html) to get the gist if not the details. Here are some thoughts on arguments and naming:

 * Use `_` in argument names (not `.`). Maybe instead of `max.tf`, use `max_tf` or maybe just `max`. `max` is probably best given that you allow for percentages in that value. 

* `procentage`? `percentage` maybe? 

* I suggest defaulting `max.words` to be `Inf` but defer to you on that. However, `tokenfilter_fun` fails on `Inf` so you'll probably want to substitute the number of works in place of `Inf`. 

* Use names that are not jargony (like `weight` instead of `K` and `weight_scheme` instead of `idf.weight`). If there are names that exist [here](https://tidymodels.github.io/model-implementation-principles/standardized-argument-names.html), please re-use them. Otherwise, we can add terms to that list so that people reuse those. 

* I don't think that you need to preface argument names with `tfidf.` since they are in that step. 

* For `tfidf`, try making the resulting variable names closer to valid R names using underscores instead of dashes, etc. You might let people pass in a naming function similar to `step_dummy`. 

* `step_hasing` use `num_terms` instead of `num`. Option to return num instead of int (for tensorflow). better naming of resulting variables. Use a suffix of "hash". Maybe call it `step_texthash` and we'll call the eventual other step `step_featurehash`. 

Other thoughts: 

 * For unit tests, it is good to have very small examples where you can inspect the data and come up with the expected results "by hand". 

* Is `idf.adjustment` a Laplace correction?

**Comment by EmilHvitfeldt** [comment]
Okay I think I got everything covered and It should be ready for a first release.

**Comment by topepo** [comment]
👍

I'll put in a PR in this week to update based on recent changes to `recipes`. I'm hoping to get `recipes` to CRAN in about 2 weeks once we get thought a [set of fixes and updates](https://github.com/orgs/tidymodels/projects/23). You should probably wait until that's on CRAN to submit `textrecipes`.

FYI we have an informal prepare-for-cran check list. [Here](https://github.com/tidymodels/yardstick/issues/57) is an example.

**Comment by EmilHvitfeldt** [comment]
Sounds good! I have added changes according to https://github.com/tidymodels/recipes/pull/215. 

Your "set of fixes and updates" link doesn't seem to work, are you referring to [recipes/upcoming release issues](https://github.com/tidymodels/recipes/issues?q=is%3Aissue+is%3Aopen+label%3A%22upcoming+release%22)?

> You should probably wait until that's on CRAN to submit `textrecipes`.

Agree, also I don't think I would even be able to since I'm depended on the dev version of `recipes`. 

With regard to `pkgdown` is that something I can set up now or would I have to wait for the move to tidymodels?

**Comment by topepo** [comment]
> Your "set of fixes and updates" link doesn't seem to work

Yikes. Some of those were unintentionally private. Try now. How many projects do you see listed on that page? 

> With regard to `pkgdown` is that something I can set up now or would I have to wait for the move to tidymodels?

I would work on the site but **don't** configure it to publish it on your `github.io` site. The `github.io` links *don't* transfer to the new organization and that has been a major pain for me.

**Comment by EmilHvitfeldt** [comment]
> I'll put in a PR in this week to update based on recent changes to recipes

Please add yourself to Authors@R

I see 25 projects right now.

**Comment by EmilHvitfeldt** [comment]
closing this issue as we are heading for the first release.
